daemons/pacemakerd/pcmkd_subdaemons.c

```diff
@@ -178,9 +177,27 @@ check_active_before_startup_processes(gpointer user_data)
             /* go to the next child and see if
                we can make progress there
              */
-            next_child++;
             break;
         case pcmk_rc_ipc_unresponsive:
+            if (!pcmk_children[next_child].respawn) {
+                /* if a subdaemon is down and we don't want it
+                   to be restarted this is a success during
+                   shutdown. if it isn't restarted anymore
+                   due to MAX_RESPAWN it is
+                   rather no success.
+                 */
+                if (pcmk_children[next_child].respawn_count <= MAX_RESPAWN) {
+                    subdaemon_check_progress = time(NULL);
+                }
+            }
+            if (!pcmk_children[next_child].active_before_startup) {
+                crm_trace("found %s[%lld] missing - signal-handler "
+                          "will take care of it",
+                           pcmk_children[next_child].name,
+                           (long long) PCMK__SPECIAL_PID_AS_0(
+                            pcmk_children[next_child].pid));
+                break;
+            }
```

Comment:
Shouldn't we do this after logging the "terminated" message below?

Reply:
You mean bailing out?
Shouldn't the signal-handler log a message with similar content?
Wanted to avoid redundant info.

Reply:
Oh right, that makes sense -- I was just looking at pcmk_process_exit() which doesn't log